stablehlo/reference/Ops.cpp

```diff
@@ -1126,6 +1143,85 @@ Tensor evalRsqrtOp(const Tensor &operand, ShapedType resultType) {
   return result;
 }
 
+SmallVector<Tensor> evalScatterOp(
+    ArrayRef<Tensor> inputs, const Tensor &scatterIndices,
+    ArrayRef<Tensor> updates, const Axes &updateWindowDims,
+    const Axes &insertedWindowDims, const Axes &scatterDimsToOperandDims,
+    Axis indexVectorDim, Region &updateComputation, Scope &scope,
+    ArrayRef<ShapedType> resultTypes) {
+  SmallVector<Tensor> results;
+  for (auto input : inputs) results.push_back(input);
+
+  Axes updateScatterDims;
+  for (auto i = 0; i < updates[0].getRank(); ++i)
+    if (!llvm::is_contained(updateWindowDims, i))
+      updateScatterDims.push_back(i);
+
+  for (auto updateIndexIt = updates[0].index_begin();
+       updateIndexIt != updates[0].index_end(); ++updateIndexIt) {
+    auto updateIndex = *updateIndexIt;
+    Sizes updateScatterIndex;
+    for (auto d : updateScatterDims)
+      updateScatterIndex.push_back(updateIndex[d]);
+
+    Index startIndex;
+    Sizes updateScatterIndexOffset(scatterIndices.getRank(), 1);
```

Comment:
sam comment as in [ here ](https://github.com/openxla/stablehlo/pull/1058/files#r1197318217)